apps/src/storage/dataBrowser/dataVisualizer/Snapshot.jsx

```diff
@@ -0,0 +1,122 @@
+/* global ClipboardItem */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import moment from 'moment';
+import msg from '@cdo/locale';
+import {html2canvas} from '@cdo/apps/util/htmlToCanvasWrapper';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import PendingButton from '@cdo/apps/templates/PendingButton';
+import * as dataStyles from '../dataStyles';
+
+class Snapshot extends React.Component {
+  static propTypes = {
+    chartTitle: PropTypes.string.isRequired,
+    selectedOptions: PropTypes.string.isRequired,
+    // Provided via Redux
+    tableName: PropTypes.string.isRequired,
+    projectName: PropTypes.string.isRequired
+  };
+
+  state = {
+    isSnapshotOpen: false,
+    isCopyPending: false,
+    isSavePending: false
+  };
+
+  handleOpen = () => this.setState({isSnapshotOpen: true});
+  handleClose = () => this.setState({isSnapshotOpen: false});
+
+  getPngBlob = () => {
+    const element = document.getElementById('snapshot');
```

Comment:
i'm not sure how much it matters in this specific instance, but best-practice is usually to use a React ref instead of `document.getElementById()` ([StackOverflow](https://stackoverflow.com/questions/37273876/reactjs-this-refs-vs-document-getelementbyid))